ext/RMagick/rmutil.c

```diff
@@ -1781,6 +1781,38 @@ rm_ensure_result(Image *image)
 }
 
 
+/**
+ * Checks if opacity_or_alpha is a named argument called alpha and returns the alpha value or
+ * converts the unnamed opacity value to alpha.
+ *
+ * No Ruby usage (internal function)
+ *
+ * @opacity_or_alpha an opacity or a named alpha value
+ * @method_name the name of the method
+ * @argument_name the name of the argument
+ */
+Quantum
+rm_get_named_alpha_value(VALUE opacity_or_alpha, const char *method_name, const char *argument_name)
+{
+    VALUE alpha;
+
+    if (TYPE(opacity_or_alpha) != T_HASH)
+    {
+        rb_warning("%s requires a named argument for '%s' and now expects an alpha value instead of an opacity value.", method_name, argument_name);
+
+        return QuantumRange - APP2QUANTUM(opacity_or_alpha);
+    }
+
+    alpha = rb_hash_aref(opacity_or_alpha, ID2SYM(rb_intern(argument_name)));
+    if (NIL_P(alpha))
+    {
+        rb_raise(rb_eArgError, "%s expects a named argument called '%s' but got a different name.", method_name, argument_name);
```

Comment:
So it looks like there are 2 possible reasons this would be `nil`. One is that they passed `alpha: nil`. The other is that they passed `something_else: 255`. I'm not sure if we need to worry about the first case, but for the second case, I think we could cover that with the typical Ruby keyword error, which looks something like `ArgumentError (missing keyword: alpha)`. Looking at the Ruby source there are the functions `rb_keyword_error` and `rb_get_kwargs`, but I'm not sure how best to use them, if at all.

Reply:
I can change the message if you would prefer that.

Reply:
I think `method_name` argument is unnecessary.
We can retrieves method name at runtime.

```
    VALUE method = rb_id2str(rb_frame_this_func());
    rb_raise(rb_eArgError, "#%"PRIsVALUE" expects a named argument called '%s' but got a different name.",  method, argument_name);
```

Reply:
That only includes the name of the function and not the name of the class. Is there also a trick to get that or are you okay with just using the function here? At the other places we provide both the class and the function.

Reply:
The class name is important things in the exception message?

Reply:
Maybe we just need it for the `rb_warning` for now but I think you are right that we don't need it in the future.

Reply:
I understand this function will be used for Image class's method only, right?

Reply:
It looks like we would only need it form the Image class but this method is generic so it would be a bit weird to have this in the utils file if we put `Image` already in there. I will move `rm_get_named_alpha_value` to the `rmimage.c` file and rename it then.